pkg/api/customization/cluster/validator.go

```diff
@@ -270,3 +276,79 @@ func (v *Validator) accessTemplate(request *types.APIContext, spec *mgmtclient.C
 
 	return nil
 }
+
+// validateGenericEngineConfig allows for additional validation of clusters that depend on Kontainer Engine or Rancher Machine driver
+func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec *v3.ClusterSpec) error {
+
+	if request.Method == http.MethodPost {
+		return nil
+	}
+
+	clusterName := request.ID
+	fmt.Printf("%v", time.Now())
+	fmt.Printf("%v", spec.AmazonElasticContainerServiceConfig)
+
+	// eks specific logic
+	prevCluster, err := v.ClusterLister.Get("", clusterName)
+	if err != nil {
+		if errors.IsNotFound(err) { //nothing to do, cluster is being created
+			return nil
+		}
+		return err
+	}
+	if spec.AmazonElasticContainerServiceConfig != nil {
+		err := validateEKS(*prevCluster.Spec.GenericEngineConfig, *spec.AmazonElasticContainerServiceConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func validateEKS(prevCluster, newCluster map[string]interface{}) error {
+	// check config is for EKS clusters
+	if driver, ok := prevCluster["driverName"]; ok {
+		if driver != service.AmazonElasticContainerServiceDriverName {
+			return nil
+		}
+	}
+
+	// don't allow for updating subnets
+	if prev, ok := prevCluster["subnets"]; ok {
+		if new, ok := newCluster["subnets"]; ok {
+			prevSlice, ok := prev.([]interface{})
+			if !ok {
+				return fmt.Errorf("unable to convert subnets to interace slice")
+			}
+			newSlice, ok := new.([]interface{})
+			if !ok {
+				return fmt.Errorf("unable to convert subnets to interface slice")
+			}
+
+			if !testEq(prevSlice, newSlice) {
+				return fmt.Errorf("cannot modify EKS subnets after creation")
```

Comment:
validators should return API errors with code, these would probably be "invalid body content"

Reply:
also, this is probably better but should note that if we were using norman tags (ik they can't be for this) I believe the fields would just be dropped and the update would go through